pkg/apitype/events.go

```diff
@@ -40,9 +40,17 @@ type DiagnosticEvent struct {
 	Message string `json:"message"`
 	Color   string `json:"color"`
 	// Severity is one of "info", "info#err", "warning", or "error".
-	Severity  string `json:"severity"`
-	StreamID  int    `json:"streamID,omitempty"`
-	Ephemeral bool   `json:"ephemeral,omitempty"`
+	Severity        string           `json:"severity"`
+	StreamID        int              `json:"streamID,omitempty"`
+	Ephemeral       bool             `json:"ephemeral,omitempty"`
+	PolicyViolation *PolicyViolation `json:"policyViolation,omitempty"`
+}
+
+// PolicyViolation represents the information related to a Policy violation.
+type PolicyViolation struct {
+	PolicyPack       PolicyPack       `json:"policyPack"`
```

Comment:
Would it make more sense to do a `PolicyIdentifier` where the policy pack name/version and policy name are all encoded?
For example, "aws-security@v5@my-awesome-rule".

Reply:
And since that stutters maybe its just "PolicyViolation.Identifier"

Reply:
That would probably be for the best. Although rather than invent our own encoding, we should just treat this as a standard URI. e.g. `ppv://<poilcy-name>/<version>/<policy>`. The benefit there is that other tools can easily parse it.